dev-tools/mage/crossbuild.go

```diff
@@ -241,9 +241,7 @@ func CrossBuildImage(platform string) (string, error) {
 	case platform == "linux/s390x":
 		tagSuffix = "s390x-debian10"
 	case strings.HasPrefix(platform, "linux"):
-		// Use an older version of libc to gain greater OS compatibility.
-		// Debian 8 uses glibc 2.19.
-		tagSuffix = "main-debian8"
+		tagSuffix = "main-debian10"
```

Comment:
Looks like the main reason for using the older Debian was to build with an older version of libc.  Perhaps we could use a newer version of Debian but also [downgrade](https://superuser.com/a/861821) glibc to 2.19 (not sure if this is possible)?  If possible, I think the downgrade would happen somewhere in https://github.com/elastic/golang-crossbuild/blob/main/go/base/Dockerfile.tmpl.

Reply:
If this works https://github.com/elastic/beats/pull/34922 I'm going to close this PR. The risk is high, as you mentioned.

Reply:
cc @andrewkroh , I recall you mentioned something in the [past](https://github.com/elastic/beats/pull/23872) about backward compatibility, do you happen to know whether this PR could cause any issues? Or the underline problem was already solved?

Reply:
This PR did cause similar problems. I detailed it in https://github.com/elastic/beats/issues/34938#issuecomment-1485646966 and I fixed the Packetbeat build. I added tests such that we now have tests in place that check that {File,Audit,Packet,Metric}beat are all compatible with glibc version shipped with RHEL 7.